snowpack/src/build/build-pipeline.ts

```diff
@@ -173,26 +174,34 @@ async function runPipelineTransformStep(
         logger.debug(`✔ transform() success [${debugPath}]`, {name: step.name});
         if (typeof result === 'string' || Buffer.isBuffer(result)) {
           // V2 API, simple string variant
-          output[destExt].code = result;
-          output[destExt].map = undefined;
+          if (destBuildFile.code !== result) {
+            destBuildFile.map = undefined;
+            if (config.buildOptions.sourceMaps && (destExt === '.js' || destExt === '.css')) {
+              logger.warn(`source-map file not found in [${debugPath}]`, {name: step.name});
+            }
+          }
+          destBuildFile.code = result;
         } else if (
           result &&
           typeof result === 'object' &&
           (result as PluginTransformResult).contents
         ) {
           // V2 API, structured result variant
-          output[destExt].code = (result as PluginTransformResult).contents;
+          destBuildFile.code = (result as PluginTransformResult).contents;
           const map = (result as PluginTransformResult).map;
           let outputMap: string | undefined = undefined;
-          if (map && config.buildOptions.sourceMaps) {
+          if (config.buildOptions.sourceMaps) {
             // if source maps disabled, don’t return any
-            if (output[destExt].map) {
-              outputMap = await composeSourceMaps(filePath, output[destExt].map!, map);
-            } else {
+            if (map) {
               outputMap = typeof map === 'object' ? JSON.stringify(map) : map;
+              if (destBuildFile.map) {
+                outputMap = await composeSourceMaps(filePath, destBuildFile.map, outputMap);
```

Comment:
Do the sourcemaps need to be composed?
I would assume that that since the plugin has access to the sourcemap, it would potentially compose the maps.